### PR TITLE
Caption next updater

### PR DESCRIPTION
**Pull request by gielcobben**

### CODE REVIEW

File: main/updater.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+const { BrowserWindow, dialog, ipcMain } = require("electron");
+const { autoUpdater } = require("electron-updater");
+const { showProgressWindow } = require("./progress");
+
+const sendStatusToWindow = text => {
+  const mainWindow = global.windows.mainWindow;
```
Comment (vernondegoede) [inline main/updater.js]: Please use object destructuring here.

File: renderer/pages/progress.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    this.state = {
+      total: 0,
+      transferred: 0,
+      percent: 0
+    };
+  }
+
+  componentDidMount() {
+    ipcRenderer.on("progress", (event, progressObject) => {
+      this.setState(progressObject);
```
Comment (vernondegoede) [inline renderer/pages/progress.js]: Don't set `progressObject` in your state directly. You only care about certain values in the `progressObject` object.

A better approach would be:

```
ipcRenderer.on("progress", (event, { transferred, total, percent }) => {
    this.setState({
        percent,
        total,
        transferred,
   });
});
```

File: renderer/pages/progress.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    });
+  }
+
+  render() {
+    const { total, delta, transferred, percent, bytesPerSecond } = this.state;
```
Comment (vernondegoede) [inline renderer/pages/progress.js]: `bytesPerSecond`, `total` and `delta` are not in your initial state. Also, you're not using them anywhere.

File: renderer/pages/start.js [hunk 4]
```diff
@@ -64,6 +64,10 @@ class MainApp extends Component {
 
     ipcRenderer.send("getStore", "language");
 
+    ipcRenderer.on("messageFromMain", (event, text) => {
```
Comment (vernondegoede) [inline renderer/pages/start.js]: What are you using this for?

File: main/updater.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  //   "Error: ",
+  //   error == null ? "unknown" : (error.stack || error).toString()
+  // );
+});
+
+autoUpdater.on("download-progress", progressObj => {
+  const progressWindow = global.windows.progressWindow;
+  progressWindow.webContents.send("progress", progressObj);
+
+  // let log_message = "Download speed: " + progressObj.bytesPerSecond;
```
Comment (vernondegoede) [inline main/updater.js]: Comment can be removed.

File: main/updater.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    };
+
+    dialog.showMessageBox(null, options);
+  }
+});
+
+autoUpdater.on("error", (event, error) => {
+  console.log(error);
+  // sendStatusToWindow("Error in auto-updater.");
+  // dialog.showErrorBox(
```
Comment (vernondegoede) [inline main/updater.js]: Comment can be removed.

File: main/updater.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+const sendStatusToWindow = text => {
+  const mainWindow = global.windows.mainWindow;
+  mainWindow.webContents.send("messageFromMain", text);
+};
+
+ipcMain.on("cancelUpdate", event => {
+  cancelUpdater();
+});
+
+autoUpdater.allowPrerelease = true;
```
Comment (vernondegoede) [inline main/updater.js]: It would be better to import the `isDev` module and use `autoUpdater.allowPrerelease = isDev;`.

File: main/updater.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+const cancelUpdater = () => {
+  const progressWindow = global.windows.progressWindow;
+
+  global.updater.cancellationToken.cancel();
+  progressWindow.hide();
+};
+
+const checkForUpdates = async () => {
+  const checking = await autoUpdater.checkForUpdates();
```
Comment (vernondegoede) [inline main/updater.js]: Just use `await autoUpdater.checkForUpdates()` since you're not using `checking` anywhere.

Comment (gielcobben) [reply main/updater.js]: Please check line `79`.

**Pull request merged**